static/src/javascripts/projects/common/modules/ui/autoupdate.js

```diff
@@ -25,190 +34,189 @@ define([
-            // there are none)
-            var identity = function (val) { return val; };
-            $(options.attachTo.filter(identity)).addClass('autoupdate--has-animation');
+                    // Insert new blocks and animate
+                    $liveblogBody.prepend(elementsToAdd);
 
-            detect.initPageVisibility();
+                    if (detect.pageVisible()) {
+                        revealInjectedElements();
+                    }
 
-            mediator.on('modules:detect:pagevisibility:visible', function () {
-                if (this.isUpdating) {
-                    this.on(); // reset backoff
-                    that.view.revealNewElements();
-                }
-            }.bind(this));
+                    toastButtonRefresh(0);
 
-            mediator.on('modules:notificationbar:show', this.view.revealNewElements.bind(this));
+                    mediator.emit('modules:autoupdate:updates', elementsToAdd.length);
 
-            // add the component to the page, and show it
-            $('.update').html(this.template).removeClass('u-h');
+                    latestBlockId = $('.block').first().attr('id');
 
-            this.btns = $(options.btnClass);
+                    newBlocks = '';
 
-            this.btns.each(function (btn) {
-                bean.add(btn, 'click', function (e) {
-                    e.preventDefault();
-                    that.view.toggle.call(that, this);
+                    RelativeDates.init();
+                    twitter.enhanceTweets();
+                }).then(function () {
+                    updating = false;
                 });
-            });
+            }
+        };
 
-            this.view.toggle.call(this, this.btns[1]);
+        var revealInjectedElements = function () {
+            fastdom.write(function () {
+                $('.autoupdate--hidden', $liveblogBody).addClass('autoupdate--highlight').removeClass('autoupdate--hidden');
+                mediator.emit('modules:autoupdate:unread', 0);
+            });
         };
 
-    }
+        //
+        // init
+        //
 
-    return Autoupdate;
+        var options = assign({
+            'toastOffsetTop': 12, // pixels from the top
+            'minUpdateDelay': (detect.isBreakpoint({ min: 'desktop' }) ? 10 : 30) * 1000, // 10 or 30 seconds minimum, depending on breakpoint
+            'maxUpdateDelay': 20 * 60 * 1000, // 20 mins
+            'backoffMultiplier': 0.75 // increase or decrease the back off rate by modifying this
+        }, opts);
 
+        var newBlocks;
+        var currentUpdateDelay = options.minUpdateDelay;
+        var latestBlockId;
+        var updating = false;
+
+        // Cache selectors
+        var $liveblogBody = $('.js-liveblog-body');
+        var $toastButton = $('.toast__button');
+        var $toastSpaceReserver = $('.toast__space-reserver');
+        var $toastText = $('.toast__text', this.$toastButton);
```

Comment:
This PR is actually just moving the contents of `newAutoupdate.js` instead `autoupdate.js`, to replace the previous system. So I haven't made any changes to the code per se, just renamed and deleted bits.

Not sure why `this` is here :confused: